[Dev Crash]: any runData will always crash with Obscure Tooltips present

**Issue:**
Any project including Obscure Tooltips in their dev env trying to run datagen will always crash, forcing you to manually swap to `compileOnly` before running datagen and then back to `implementation` to include it during runtime

**To reproduce:**
- Open up a dev instance of Obscure Tooltips
- run the `runData` gradle task
-> task will crash and fail

```java
[00:32:41] [main/FATAL] [ne.ne.fm.ModLoader/LOADING]: Failed to wait for future Mod Construction, 1 errors found
Exception in thread "main" java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:32)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.Launcher.run(Launcher.java:103)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.Launcher.main(Launcher.java:74)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23)
	at cpw.mods.bootstraplauncher@2.0.2/cpw.mods.bootstraplauncher.BootstrapLauncher.run(BootstrapLauncher.java:210)
	at cpw.mods.bootstraplauncher@2.0.2/cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:69)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at net.neoforged.devlaunch.Main.main(Main.java:57)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:118)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at MC-BOOTSTRAP/fml_loader@4.0.42/net.neoforged.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:136)
	at MC-BOOTSTRAP/fml_loader@4.0.42/net.neoforged.fml.loading.targets.CommonLaunchHandler.dataService(CommonLaunchHandler.java:132)
	at MC-BOOTSTRAP/fml_loader@4.0.42/net.neoforged.fml.loading.targets.NeoForgeDataDevLaunchHandler.runService(NeoForgeDataDevLaunchHandler.java:28)
	at MC-BOOTSTRAP/fml_loader@4.0.42/net.neoforged.fml.loading.targets.CommonLaunchHandler.lambda$launchService$4(CommonLaunchHandler.java:118)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.5/cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30)
	... 11 more
Caused by: net.neoforged.fml.ModLoadingException: Loading errors encountered:
	- Obscure Tooltips (obscure_tooltips) has failed to load correctly
	  java.lang.NullPointerException: Cannot invoke "net.minecraft.client.Minecraft.getResourceManager()" because the return value of "net.minecraft.client.Minecraft.getInstance()" is null

	at MC-BOOTSTRAP/fml_loader@4.0.42/net.neoforged.fml.ModLoader.waitForFuture(ModLoader.java:281)
	at MC-BOOTSTRAP/fml_loader@4.0.42/net.neoforged.fml.ModLoader.dispatchParallelTask(ModLoader.java:244)
	at MC-BOOTSTRAP/fml_loader@4.0.42/net.neoforged.fml.ModLoader.constructMods(ModLoader.java:152)
	at MC-BOOTSTRAP/fml_loader@4.0.42/net.neoforged.fml.ModLoader.gatherAndInitializeMods(ModLoader.java:142)
	at TRANSFORMER/neoforge@21.1.219/net.neoforged.neoforge.internal.CommonModLoader.begin(CommonModLoader.java:49)
	at TRANSFORMER/neoforge@21.1.219/net.neoforged.neoforge.data.loading.DatagenModLoader.begin(DatagenModLoader.java:47)
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.data.Main.main(Main.java:101)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	... 17 more
```

**Cause:**
Datagen will call both server and client entry points, but neither will be present at the time. Calling `Minecraft.getInstance()` during datagen will return `null`, so the following like will throw a NullPointer
https://github.com/ObscuriaLithium/obscure-tooltips/blob/ae87591dfad8de64829645269177fbf74ff1bea8/neoforge/src/main/java/dev/obscuria/tooltips/neoforge/NeoObscureTooltips.java#L17

**Solution:**
Simply do a null check before calling `getResourceManager()`.

--- a/neoforge/src/main/java/dev/obscuria/tooltips/neoforge/NeoObscureTooltips.java
+++ b/neoforge/src/main/java/dev/obscuria/tooltips/neoforge/NeoObscureTooltips.java
@@ -14,7 +14,9 @@ public final class NeoObscureTooltips {
     public NeoObscureTooltips(IEventBus eventBus) {
         if (FMLEnvironment.dist.isDedicatedServer()) return;
         ObscureTooltips.init();
-        if (Minecraft.getInstance().getResourceManager() instanceof ReloadableResourceManager manager)
+        Minecraft mc = Minecraft.getInstance();
+        if (mc == null) return;
+        if (mc.getResourceManager() instanceof ReloadableResourceManager manager)
             manager.registerReloadListener(TooltipManager.INSTANCE);
     }
 }

